pkg/rbac/rbac.go

```diff
@@ -270,13 +270,19 @@ func NewEnforceHandler(l *zap.SugaredLogger, basePath string, enforcer *casbin.E
 		if resource == ResourceNamespaces {
 			return true, nil
 		}
+		// Allow creating a restore without a name.
+		// RBAC is enforced in the individual methods.
+		if resource == ResourceDatabaseClusterRestores && name == "" && action == ActionCreate {
+			return true, nil
+		}
```

Comment:
Since we'll be doing the same for `ResourceDatabaseClusterBackups` as well, WDYT about moving the `create-able` objects to a list and checking that? Like we do below for `allowedObjectsForListing`?